analyzer/tools/build-logger/src/ldlogger-tool-gcc.c

```diff
@@ -388,9 +422,37 @@ int loggerGccParserCollectActions(
     if (strstr(toolName_, cppCompiler[i]))
       lang = CPP;
 
+  LoggerVector extendedArgs;
+  loggerVectorInit(&extendedArgs);
```

Comment:
I believe you should explicitly call `loggerVectorClear()` at the end of usage in order to avoid memory leak. This is the "destructor" of the vector type.